dask/array/core.py

```diff
@@ -575,6 +575,8 @@ def blockdims_from_blockshape(shape, blockshape):
     >>> blockdims_from_blockshape((10, 10), (4, 3))
     ((4, 4, 2), (3, 3, 3, 1))
     """
+    shape = map(int, shape)
+    blockshape = map(int, blockshape)
```

Comment:
This hides some of the informative errors below.

We might also want to genuinely raise an error if the mantissa is nonzero.